zarr/storage.py

```diff
@@ -2507,7 +2511,9 @@ def __init__(self, store, metadata_key='.zmetadata'):
                                 consolidated_format)
 
         # decode metadata
-        self.meta_store = meta['metadata']
+        self.meta_store = {}
+        for k, v in meta["metadata"].items():
+            self.meta_store[k] = json_dumps(v)
```

Comment:
FWIW this was originally how this worked, but this introduces an extra unnecessary JSON encode/decode round trip into the processing of consolidated metadata. In practice that may not be a performance issue, but it seemed awkward to have this extra JSON encode/decode if it wasn't necessary. Hence I introduced a special case which avoided the necessity for this, see the [parse_metadata function in the original PR](https://github.com/zarr-developers/zarr-python/pull/268/files#diff-30e7a6c546b5e66e78ba8f559c2b4c1d).